scripts/rabbitmq-server-ha.ocf

```diff
@@ -812,10 +812,45 @@ get_master_name_but()
     done
 }
 
+erl_eval() {
+    local fmt="${1:?}"
+    shift
+
+    ${OCF_RESKEY_ctl} eval "$(printf "$fmt" "$@")"
+}
+
 # Returns 0 if we are clustered with provideded node
 is_clustered_with()
 {
-    get_running_nodes | grep -q $(rabbit_node_name $1);
+    local LH="${LH}: is_clustered_with: "
+    local node_name
+    local rc
+    node_name=$(rabbit_node_name $1)
+
+    local seen_as_running
+    seen_as_running=$(erl_eval "lists:member('%s', rabbit_mnesia:cluster_nodes(running))." "$node_name")
```

Comment:
By approach do you mean using `lists:member/2` or checking both running and partitioned states? 

I've added `lists:member/2` because I needed to add another check and did it just for unification with new piece of code. It's slightly cleaner IMO, but it's not worth changing it everywhere. And for partition check it's just way more harder to implement it using grep.